### DISCUSSION TITLE
CAN-29 and BR-OPT-00100-0036 rule

### DISCUSSION THREAD

**Discussion by JM-Allier** [OP-TED/eForms-SDK#1038] [Q&A] [answered]
Hello 
We have an issue with the rule "BR-OPT-00100-0036" for OPT-100 in CAN-29 form.
Some of our buyer cannot publish a contrat award notice because there is no previous Framework notice.
In the case of en "exclusivity contract" ( I'm not sure of the english translation) above the threshold the buyer do not need to publish the contract notice and has no award criterion.

It is impossible to publish the contrat Award without a "NoticeDocumentReference" but our customers do not understand why. This seems to be a technical issue rather than a legal one. Our jurist confirm that whe should be able to publish this kind of notice.

How can we do ? Is there a legal fondation to the rule BR-OPT-00100-0036 ? 

Here is an example of failed assert 

`<svrl:failed-assert id="BR-OPT-00100-0036" location="/can:ContractAwardNotice/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:NoticeResult/efac:SettledContract" test="count(cac:NoticeDocumentReference/cbc:ID) > 0 or not(efbc:ContractFrameworkIndicator = true())" role="ERROR">
	<svrl:text>'Avis qui a créé l'accord-cadre' (OPT-100-Contract) est obligatoire dans le type d’avis '29 – Avis d’attribution de marché – directive générale, régime ordinaire', sauf dans les conditions suivantes: 'Le marché est attribué dans le contexte d’un accord-cadre' (BT-768-Contract) n’est pas vrai</svrl:text>
	<svrl:diagnostic-reference diagnostic="OPT-100-Contract" see="field:OPT-100-Contract">
		<svrl:text>cac:NoticeDocumentReference/cbc:ID</svrl:text>
	</svrl:diagnostic-reference>
</svrl:failed-assert>`
	
Thank you 
Jean-Marc

**Comment by JM-Allier** [comment, marked as answer]
I think I find the answer in this post 
https://github.com/OP-TED/eForms-SDK/discussions/915
We do not have to use 
`<efbc:ContractFrameworkIndicator>true</efbc:ContractFrameworkIndicator>`

in the contract award notice of the framework notice, but just for the contract within th eframework.
Jean-Marc

**Reply by bertrand-lorentz** [reply]
Yes, that is correct.
The description of the corresponding field/BT is : "The contract is awarded within a framework agreement."
